Analysis/Tasks/trackchecks.cxx

```diff
@@ -0,0 +1,225 @@
+    hDCAxyReco_truepid->Add(2, TH2F("hDCAxyReco_truepid_ka", "DCAxy reco (true Ka); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyReco_truepid->Add(3, TH2F("hDCAxyReco_truepid_pr", "DCAxy reco (true Pr); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyReco_truepid->Add(4, TH2F("hDCAxyReco_truepid_de", "DCAxy reco (true De); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+
+    hDCAxyPrim_truepid->Add(0, TH2F("hDCAxyPrim_truepid_el", "DCAxy primaries (true El); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyPrim_truepid->Add(1, TH2F("hDCAxyPrim_truepid_pi", "DCAxy primaries (true Pi); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyPrim_truepid->Add(2, TH2F("hDCAxyPrim_truepid_ka", "DCAxy primaries (true Ka); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyPrim_truepid->Add(3, TH2F("hDCAxyPrim_truepid_pr", "DCAxy primaries (true Pr); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxyPrim_truepid->Add(4, TH2F("hDCAxyPrim_truepid_de", "DCAxy primaries (true De); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+
+    hDCAxySeco_truepid->Add(0, TH2F("hDCAxySeco_truepid_el", "DCAxy secondaries (true El); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxySeco_truepid->Add(1, TH2F("hDCAxySeco_truepid_pi", "DCAxy secondaries (true Pi); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxySeco_truepid->Add(2, TH2F("hDCAxySeco_truepid_ka", "DCAxy secondaries (true Ka); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxySeco_truepid->Add(3, TH2F("hDCAxySeco_truepid_pr", "DCAxy secondaries (true Pr); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+    hDCAxySeco_truepid->Add(4, TH2F("hDCAxySeco_truepid_de", "DCAxy secondaries (true De); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+
+    fCutDCAxy = new TF1("fMaxDCAxy", "[0]+[1]/(x^[2])", 0, 1e10); //from TPC analysis task (Run2)
+    fCutDCAxy->SetParameter(0, 0.0105);
+    fCutDCAxy->SetParameter(1, 0.0350);
+    fCutDCAxy->SetParameter(2, 1.1);
+  } //init
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& col, soa::Join<aod::Tracks, aod::TracksExtra, aod::McTrackLabels>& tracks, aod::McParticles& mcParticles)
+  {
+
+    if (isMC) {
+      LOGF(info, "isMC=%d - Running on MC", isMC.value);
+    } else {
+      LOGF(info, "isMC=%d - Running on DATA", isMC.value);
+    }
+
+    //event selection
+    if (!isMC && !col.alias()[kINT7]) // trigger (should be skipped in MC)
+      return;
+    if (!col.sel7()) //additional cuts
+      return;
+    if (abs(col.posZ()) > 10.) // |v_z|<10cm
```

Comment:
You can use a Filter for this

Reply:
I put the vertex_z selection into a filter. I do not know how to put the others cuts, I tried different possibilities but it doesn't compile.